crates/snapbox/src/data.rs

```diff
@@ -96,35 +122,55 @@ impl Data {
         match &self.inner {
             DataInner::Binary(_) => None,
             DataInner::Text(data) => Some(data.to_owned()),
+            #[cfg(feature = "json")]
+            DataInner::Json(value) => Some(serde_json::to_string_pretty(value).unwrap()),
         }
     }
 
     pub fn to_bytes(&self) -> Vec<u8> {
         match &self.inner {
             DataInner::Binary(data) => data.clone(),
             DataInner::Text(data) => data.clone().into_bytes(),
+            #[cfg(feature = "json")]
+            DataInner::Json(value) => serde_json::to_vec_pretty(value).unwrap(),
         }
     }
 
     pub fn try_coerce(self, format: DataFormat) -> Self {
```

Comment:
As we get to tests, we should ensure
- to_bytes/render produce the same results
- all types are coercible to each other
- coercible conversions create the same output as `to_bytes`/`render`
- Normalization on json

Reply:
This [URL](https://github.com/assert-rs/trycmd/compare/2bc7f810ef4d8d1226d200ada97cc0d154ebe623..3ed730d5dd22244daf2eba2318b8a0cf45bd01cc) should show the diff between when this was posted and what was changed to add tests. How I added the `NormalizeMatches` support for JSON messed up the diff for those changes (that commit will be squashed but I wanted it to be easier to review, [diff here](https://github.com/assert-rs/trycmd/compare/3ed730d5dd22244daf2eba2318b8a0cf45bd01cc..f53cd76a672fb27d06208e2ecd9eebab844a0f18)).